src/act-runner/runTestCases.ts

```diff
@@ -0,0 +1,56 @@
+import * as fs from 'fs'
+import * as path from 'path';
+import { Config, TestCase, Assertion, EarlReport } from '../types';
+import { groupedTestCases, loadTestCases } from './groupTestCases';
+import { runRuleTests, PageRunner } from './runRuleTests';
+
+export const runTestCases = async (
+  config: Config, 
+  pageRunner: PageRunner
+): Promise<EarlReport[]> => {
+  const { testCaseJson, ruleId } = config;
+  const cacheResults = getCacheResults(config);
+  const testCases = await loadTestCases(testCaseJson);
+  const tests = groupedTestCases(testCases, ruleId);
+  const results = [];
+
+  for (let testcases of tests) {
+    const { ruleName, ruleId } = testcases[0];
+    console.log(`Testing: ${ruleName} (${ruleId})`);
+    process.stdout.write("  ");
+
+    const untestedCases: TestCase[] = [];
+    testcases.forEach(testcase => {
+      const testcaseCache = cacheResults
+        .filter(({ subject }) => {
+          return subject.source.includes(testcase.relativePath);
+        })
+        .map(assertion => ({ "@graph": [assertion] }));
+
+      if (testcaseCache.length > 0) {
+        results.push(...testcaseCache);
+      } else {
+        untestedCases.push(testcase);
+      }
+    });
+    // Show cached results
+    process.stdout.write(",".repeat(testcases.length - untestedCases.length));
+
+    try {
+      const ruleResults = await runRuleTests(pageRunner, untestedCases);
+      if (ruleResults && ruleResults.length > 0) {
+        results.push(...ruleResults);
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  }
+  return results;
+};
+
+function getCacheResults({ outFile }: Config): Assertion[] {
+  if (!fs.existsSync(outFile)) {
```

Comment:
When does the test result get written to a file so it then is cached? Can't seem to find that code.

Reply:
Here: https://github.com/dequelabs/act-reports-axe/pull/1/files/b0be91b00655bb76151af7572dcebe42265d3561#diff-64432cee3c436da626010144faad5a49d95a454f52e9e49e47b16cc09d513096R16

So you run first, and then any time you rerun with the same outFile it'll pull from cache what it can. I'm not sure I actually still need this, I had this i here to speed up a build process that relied on running axe-core a thousand times.